### PR TITLE
Unit test code coverage

### LINKED ISSUE #18: add code coverage reports in CI
We need unit test code coverage reports in our CI. In kubevirt we use [coveralls](https://coveralls.io/) to achieve this. There might be a better option that integrates with prow.

- investigate code coverage tools
- introduce code coverage for unit tests as part of PR ci. 
- Revise CONTRIBUTING.md to reflect our goals for code coverage changes and define expectations

### PR DESCRIPTION
**Pull request by slintes**
Added unit test code coverage using drone.io and coveralls.io

In order to prevent that the build tool image needs to be built for every PR change, it is built on merges only.

Closes #18 

/hold
Let's see if it works...

### CODE REVIEW

**Comment by slintes** [comment]
works now, ready for review and can go in after #198 is merged

**Comment by fromanirh** [comment]
big plus for the idea, I'll review later

File: Makefile [hunk 1]
```diff
@@ -22,6 +22,12 @@ FULL_REGISTRY_IMAGE ?= "${IMAGE_REGISTRY}/${REGISTRY_NAMESPACE}/${REGISTRY_IMAGE
 
 CLUSTER ?= "ci"
 
+GIT_VERSION=$$(git describe --always --tags)
+VERSION=$${CI_UPSTREAM_VERSION:-$(GIT_VERSION)}
+GIT_COMMIT=$$(git rev-list -1 HEAD)
+COMMIT=$${CI_UPSTREAM_COMMIT:-$(GIT_COMMIT)}
+BUILD_DATE=$$(date --utc -Iseconds)
```
Comment (fromanirh) [inline Makefile:29]: nice, TIL!

**Pull request merged**